internal/store/store.go

```diff
@@ -734,6 +742,42 @@ func (s *Store) UpdateImageBuildInCompose(composeID uuid.UUID, imageBuildID int,
 	})
 }
 
+func (s *Store) AddImageToImageUpload(composeID uuid.UUID, imageBuildID int, reader io.Reader) error {
+	currentCompose, exists := s.Composes[composeID]
+	if !exists {
+		return &NotFoundError{"compose does not exist"}
+	}
+
+	imageBuild := currentCompose.ImageBuilds[imageBuildID]
+	localTarget := imageBuild.GetLocalTarget()
+
+	if localTarget == nil {
+		return &NoLocalTargetError{fmt.Sprintf("image upload requested for compse %s and image build %d but it has no local target", composeID.String(), imageBuildID)}
```

Comment:
I like that you make sure to error out in this case, but a thought: wouldn't it be nice to treat this error the same way as if the endpoint was invalid? That way, if in the future we dropped the local target completely, the behaviour from the point of view of a worker that never received jobs with the local target set would be exactly the same.

Reply:
So what do you propose? I'm probably missing out what is a better solution from your point of view.

Reply:
My idea was to make sure this would return 404. Let's discuss that separately, this is good to go :)